spec/expectations/images.js

```diff
@@ -163,6 +163,121 @@ exports.getExpectedImageAttributes = (image, options = {}) => {
   return compactObject(expected);
 };
 
+
+/**
+ * Builds expected image attributes from the PUT/POST image request
+ * Returns the expected image attributes JSON from the API for a given database row.
+ *
+ * @param {Object} request - The PUT/POST request
+ * @param {Object} results - The results from the GET request to compare.
+ * @param {Object} [options] - Additional attributes not available in the request (id and owner_id from image).
+ * @returns {Object} The expected API correction.
+ */
+exports.getExpectedRequestedImageAttributes = (request, results, options) => {
+
+  const { is_published, caption, link, download_link, shop_link, view_type, 
+    date_shot, date_shot_min, date_shot_max
+  } = request.body;
+  
+  const requestBody = request.body;
+  const resultsBody = results.body;
```

Comment:
you cannot take expected values from the result returned since it is the one we want to compare to

Reply:
you should take those values from the test context where you run it probably or a previous GET maybe in case of PUT